src/test/java/net/snowflake/client/jdbc/ConnectionLatestIT.java

```diff
@@ -1379,6 +1380,86 @@ public void testDataSourceOktaGenerates429StatusCode() throws Exception {
+    assertEquals(
+        "true", props.get(SFSessionProperty.JDBC_ARROW_TREAT_DECIMAL_AS_INT.getPropertyKey()));
+    assertEquals("true", props.get(SFSessionProperty.ALLOW_UNDERSCORES_IN_HOST.getPropertyKey()));
+    assertEquals(
+        "/some/path/file.json", props.get(SFSessionProperty.CLIENT_CONFIG_FILE.getPropertyKey()));
+    assertEquals(
+        "false", props.get(SFSessionProperty.DISABLE_GCS_DEFAULT_CREDENTIALS.getPropertyKey()));
+    assertEquals("false", props.get(SFSessionProperty.DISABLE_SAML_URL_CHECK.getPropertyKey()));
+    assertEquals("false", props.get(SFSessionProperty.DISABLE_SOCKS_PROXY.getPropertyKey()));
+    assertEquals("true", props.get(SFSessionProperty.ENABLE_PATTERN_SEARCH.getPropertyKey()));
+    assertEquals("DB_NAME", props.get(SFSessionProperty.DATABASE.getPropertyKey()));
+    assertEquals("false", props.get(SFSessionProperty.ENABLE_PUT_GET.getPropertyKey()));
+    assertEquals("5", props.get(SFSessionProperty.MAX_HTTP_RETRIES.getPropertyKey()));
+    assertEquals("10", props.get(SFSessionProperty.NETWORK_TIMEOUT.getPropertyKey()));
+    assertEquals("false", props.get(SFSessionProperty.OCSP_FAIL_OPEN.getPropertyKey()));
+    assertEquals("proxyHost.com", props.get(SFSessionProperty.PROXY_HOST.getPropertyKey()));
+    assertEquals("8080", props.get(SFSessionProperty.PROXY_PORT.getPropertyKey()));
+    assertEquals("http", props.get(SFSessionProperty.PROXY_PROTOCOL.getPropertyKey()));
+    assertEquals("proxyUser", props.get(SFSessionProperty.PROXY_USER.getPropertyKey()));
+    assertEquals("proxyPassword", props.get(SFSessionProperty.PROXY_PASSWORD.getPropertyKey()));
+    assertEquals("3", props.get(SFSessionProperty.PUT_GET_MAX_RETRIES.getPropertyKey()));
+    assertEquals("true", props.get(SFSessionProperty.STRINGS_QUOTED.getPropertyKey()));
+
+    ds.setOauthToken("a_token");
+    assertEquals("OAUTH", props.get(SFSessionProperty.AUTHENTICATOR.getPropertyKey()));
+    assertEquals("a_token", props.get(SFSessionProperty.TOKEN.getPropertyKey()));
+
+    ds.setPasscodeInPassword(true);
+    assertEquals("true", props.get(SFSessionProperty.PASSCODE_IN_PASSWORD.getPropertyKey()));
+    assertEquals(
+        "USERNAME_PASSWORD_MFA", props.get(SFSessionProperty.AUTHENTICATOR.getPropertyKey()));
+
+    ds.setPrivateKeyFile("key.p8", "pwd");
+    assertEquals("key.p8", props.get(SFSessionProperty.PRIVATE_KEY_FILE.getPropertyKey()));
+    assertEquals("pwd", props.get(SFSessionProperty.PRIVATE_KEY_FILE_PWD.getPropertyKey()));
+    assertEquals("SNOWFLAKE_JWT", props.get(SFSessionProperty.AUTHENTICATOR.getPropertyKey()));
+
+    ds.setPasscodeInPassword(false);
+    ds.setPasscode("a_passcode");
+    assertEquals("false", props.get(SFSessionProperty.PASSCODE_IN_PASSWORD.getPropertyKey()));
+    assertEquals(
+        "USERNAME_PASSWORD_MFA", props.get(SFSessionProperty.AUTHENTICATOR.getPropertyKey()));
+    assertEquals("a_passcode", props.get(SFSessionProperty.PASSCODE.getPropertyKey()));
```

Comment:
let's add also diagnostic options added in https://github.com/snowflakedb/snowflake-jdbc/pull/1789

Reply:
Done.